packages/@ama-sdk/schematics/schematics/typescript/core/index.ts

```diff
@@ -36,119 +66,189 @@ const getPathObjectTemplate = (pathObj: PathObject) => {
-        };
-        return operation;
-      }) as Operation[]
-    }));
-    return extraction || [];
-  };
 
-  /**
-   * rule to clear previous SDK generation
-   * @param tree
-   * @param _context
-   */
-  const clearGeneratedCode = (tree: Tree, _context: SchematicContext) => {
-    treeGlob(tree, path.posix.join(targetPath, 'src', 'api', '**', '*.ts')).forEach((file) => tree.delete(file));
-    treeGlob(tree, path.posix.join(targetPath, 'src', 'models', 'base', '**', '!(index).ts')).forEach((file) => tree.delete(file));
-    treeGlob(tree, path.posix.join(targetPath, 'src', 'spec', '!(operation-adapter|index).ts')).forEach((file) => tree.delete(file));
-    return tree;
-  };
+    const generateOperationFinder = async (): Promise<PathObject[]> => {
+      const swayOptions = {
+        definition: path.resolve(generatorOptions.specPath!)
+      };
+      const swayApi = await sway.create(swayOptions);
+      const extraction = swayApi.getPaths().map((obj) => ({
+        path: `${obj.path as string}`,
+        regexp: obj.regexp as RegExp,
+        operations: obj.getOperations().map((op: any) => {
+          const operation: Operation = {
+            method: `${op.method as string}`,
+            operationId: `${op.operationId as string}`
+          };
+          return operation;
+        }) as Operation[]
+      }));
+      return extraction || [];
+    };
 
-  /**
-   * rule to update readme and generate mandatory code source
-   */
-  const generateSource = async () => {
-    if (!existsSync(specPath)) {
-      throw new Error(`${specPath} does not exist`);
-    }
+    /**
+     * rule to update readme and generate mandatory code source
+     */
+    const generateSource = async () => {
+      const pathObjects = await generateOperationFinder();
+      const swayOperationAdapter = `[${pathObjects.map((pathObj) => getPathObjectTemplate(pathObj)).join(',')}]`;
 
-    const pathObjects = await generateOperationFinder();
-    const swayOperationAdapter = `[${pathObjects.map((pathObj) => getPathObjectTemplate(pathObj)).join(',')}]`;
-
-    return mergeWith(apply(url('./templates'), [
-      template({
-        ...options,
-        swayOperationAdapter,
-        empty: ''
-      }),
-      move(targetPath),
-      renameTemplateFiles()
-    ]), MergeStrategy.Overwrite);
-  };
+      return mergeWith(apply(url('./templates'), [
+        template({
+          ...options,
+          swayOperationAdapter,
+          empty: ''
+        }),
+        move(targetPath),
+        renameTemplateFiles()
+      ]), MergeStrategy.Overwrite);
+    };
+
+    /**
+     * Update local swagger spec file
+     */
+    const updateSpec = () => {
+      const readmeFile = path.posix.join(targetPath, 'readme.md');
+      const specContent = readFileSync(generatorOptions.specPath!).toString();
+      if (tree.exists(readmeFile)) {
+        const swaggerVersion = /version: ([0-9]+\.[0-9]+\.[0-9]+)/.exec(specContent);
```

Comment:
The variable name is not really correct, it is not `swaggerVersion` here, it is more the `specVersion`.
I also think that your request may not handle pre-release versions:
```suggestion
        const swaggerVersion = /version: *([0-9]+\.[0-9]+\.[0-9]+(?:-[^ ]+)?)/.exec(specContent);
```